### PR TITLE
Add Blake2b

### PR DESCRIPTION
**Pull request by bbedward**
This is a rebase of #113.

I'm using this for a flutter crypto currency wallet I'm working on, as seen implemented in this package: https://github.com/bbedward/flutter_nano_core

### CODE REVIEW

**Comment by stevenroose** [comment]
Hmm, it'd be nice if this PR could work on top of the commits from #113. Both to give credit to the original author and to have the name of the implementer on top of the commit in case someone has questions about it later.

**Comment by stevenroose** [comment]
In that case, I could merge this next week and put it into a release as well.

**Comment by bbedward** [comment]
I'll rebase it soon, originally it was just for personal usage but I'll redo it to preserve the original commits

**Comment by stevenroose** [comment]
Thanks!

**Comment by bbedward** [comment]
@stevenroose Re-based this pull request to include @jminer's commit

File: test/test/digest_tests.dart [hunk 1]
```diff
@@ -31,5 +33,14 @@ void _runDigestTest(
   var out = digest.process(plainText);
   var hexOut = formatBytesAsHexString(out);
 
+  expect(hexOut, equals(expectedHexDigestText));
+
+ f or(var i = 0; i < plainText.length; ++i) {
```
Comment (stevenroose) [inline test/test/digest_tests.dart]: There is a type on this line.

**Review by stevenroose — CHANGES_REQUESTED** [review]

File: test/test/digest_tests.dart (hunk 1 above)
Comment (bbedward) [reply test/test/digest_tests.dart]: Pushed fix,

also confirmed test passes

```
00:00 +31: Blake2b: digest: (empty string)
00:00 +32: Blake2b: digest: Lorem ipsum dolor sit amet[...]
00:00 +33: Blake2b: digest: En un lugar de La Mancha, [...]
00:00 +34: Blake2b: digest: Lorem ipsum dolor sit amet[...]
00:00 +35: Blake2b: digest: Lorem ipsum dolor sit amet[...]
```

**Comment by stevenroose** [comment]
Thanks!

**Pull request merged**